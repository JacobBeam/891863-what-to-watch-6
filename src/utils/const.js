export const TabType = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const RatingLevels = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`
};
