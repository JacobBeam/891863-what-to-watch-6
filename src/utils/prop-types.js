import PropTypes from 'prop-types';

export const filmsPropTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    posterImage: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    id: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewVideoLink: PropTypes.string.isRequired,
  })).isRequired,
};

export const filmPropTypes = {
  film: PropTypes.shape({
    name: PropTypes.string.isRequired,
    posterImage: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    id: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewVideoLink: PropTypes.string.isRequired,
  }).isRequired,
};
