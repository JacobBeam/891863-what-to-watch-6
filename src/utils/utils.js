import {RatingLevels, allGenreFilter} from './const';

export const findFilmById = (films, id) => {
  return films.find((element) => element.id === id);
};


export const conversionMinutes = (mins) => {
  const hours = Math.trunc(mins / 60);
  const minutes = mins % 60;
  return `${hours}h ${minutes}m`;
};


export const findRatingLevel = (rating) => {

  let currentRating = ``;

  switch (true) {
    case (rating >= 0 && rating < 3):
      currentRating = RatingLevels.BAD;
      break;
    case (rating >= 3 && rating < 5):
      currentRating = RatingLevels.NORMAL;
      break;
    case (rating >= 5 && rating < 8):
      currentRating = RatingLevels.GOOD;
      break;
    case (rating >= 5 && rating < 10):
      currentRating = RatingLevels.VERY_GOOD;
      break;
    case (rating >= 10):
      currentRating = RatingLevels.AWESOME;
      break;
  }
  return currentRating;
};

export const filterFilmsByGenre = (genre, films) => {

  if (genre === allGenreFilter.ALL_GENRES) {
    return films;
  }

  return films.filter((film)=>film.genre === genre);
};
