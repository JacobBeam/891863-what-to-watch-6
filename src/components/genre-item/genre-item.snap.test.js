import React from 'react';
import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Router} from 'react-router-dom';
import GenreItem from './genre-item';
import {allGenreFilter} from '../../utils/const';

const api = jest.fn();
const middlleware = thunk.withExtraArgument(api);
const mockStore = configureStore([middlleware]);
const name = allGenreFilter.ALL_GENRES;
const onResetCountFilms = jest.fn();

it(`Should GenreItem render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore({
    FILM_DATA: {genre: `Test`},
  });

  const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <GenreItem name={name} onResetCountFilms={onResetCountFilms}/>
        </Router>
      </redux.Provider>
  );
  expect(container).toMatchSnapshot();
});
