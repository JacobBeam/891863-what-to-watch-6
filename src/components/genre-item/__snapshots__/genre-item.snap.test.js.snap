// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GenreItem render correctly 1`] = `
<div>
  <li
    class="catalog__genres-item "
  >
    <a
      class="catalog__genres-link"
      href="#"
    >
      All genres
    </a>
  </li>
</div>
`;
